cluster.go

```diff
@@ -284,17 +284,41 @@ func (c *Cluster) setState(state string) {
 		return
 	}
 
+	c.logger().Printf("Change cluster state from %s to %s", c.State, state)
+
+	var doCleanup bool
+
 	switch state {
 	case ClusterStateResizing:
 		c.prefect.SetRestricted()
 	case ClusterStateNormal:
 		c.prefect.SetNormal()
 		// Don't change routing for these states:
 		// - ClusterStateStarting
+
+		// If state is RESIZING -> NORMAL then run cleanup.
+		if c.State == ClusterStateResizing {
+			doCleanup = true
+		}
 	}
 
-	c.logger().Printf("Change cluster state from %s to %s", c.State, state)
 	c.State = state
+
+	// TODO: consider NOT running cleanup on an active node that has
+	// been removed.
+	// It's safe to do a cleanup after state changes back to normal.
+	if doCleanup {
+		var cleaner HolderCleaner
+		cleaner.URI = c.URI
+		cleaner.Holder = c.Holder
+		cleaner.Cluster = c
+		cleaner.Closing = c.closing
+
+		// Clean holder.
+		if err := cleaner.CleanHolder(); err != nil {
```

Comment:
Im not sure of the correct answer here, but i'm not sure if we want to leave it in state NORMAL if the cleanup fails, though I'm not clear on what else it could be

Reply:
This is a valid concern.

I think the biggest problem would be if you resized a cluster, the cleanup failed, and then later you did another resize and those lingering fragments were  conflicting with the new topology. We need to test that scenario and see how a resize over orphaned fragments behaves.

Another option is to hold the cluster in an inactive state until manual intervention.